src/Examine.AzureDirectory/AzureReadOnlyDirectory.cs

```diff
@@ -0,0 +1,335 @@
+﻿using Examine.LuceneEngine.Directories;
+
+namespace Examine.AzureDirectory
+{
+  using System;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using Lucene.Net.Index;
+using Lucene.Net.Store;
+using Microsoft.Azure.Storage;
+using Microsoft.Azure.Storage.Blob;
+using Directory = Lucene.Net.Store.Directory;
+
+namespace Examine.AzureDirectory
+{
+    public class AzureReadOnlyDirectory : ExamineDirectory
```

Comment:
This class seems like a huge amount of duplication why is this necessary as opposed to the isReadOnly flag in the normal AzureDirectory class?